db/version_set.cc

```diff
@@ -1534,6 +1534,35 @@ void Version::GetCreationTimeOfOldestFile(uint64_t* creation_time) {
   *creation_time = oldest_time;
 }
 
+Status Version::TryVerifySstUniqueIds() {
+  for (int level = 0; level < storage_info_.num_non_empty_levels_; level++) {
+    for (FileMetaData* meta : storage_info_.LevelFiles(level)) {
+      // if table is not opened or manifest doesn't have unique_id, skip
```

Comment:
Given that this feature is meant to be a cheaper replacement for the VerifyFileChecksums API, and the latter verifies all files and not just ones with opened table readers, I wonder if we should create a temporary table_reader if it doesn't have one already.

Reply:
I'm planning to enable the feature by default after a few release and then removing the option to enforce the verification, as there's limited performance overhead for this feature and unique_id is always tracked in manifest anyway (with or without enabling this). That's why it doesn't explicitly try opening more files than the current user setting (plus `max_open_files` is default to -1 to open all files). Also we introduced a statistics `NUMBER_SST_UNIQUE_ID_VERIFICATION_SKIP` to let the user know if they have skipped verification.
And for zippy, they have the default setting for `max_open_files = -1`.

Reply:
I'm worried that this might leave a hole in the verification (when max_open_files != -1). In that case, is the unique ID verified when a file is eventually opened?

Reply:
Sounds good, updated to force verifying all SST files and removed statistics.